pkg/clients/edgex-foundry/util.go

```diff
@@ -44,20 +44,25 @@ type ClientURL struct {
 	Port int
 }
 
-func getEdgeDeviceName(d *devicev1alpha1.Device) string {
+type EdgeNameProvider interface {
```

Comment:
Maybe you don't need to define a separate interface and use [metav1.Object](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L33) instead. The methods GetLabels() and GetName() have been defined there.

Reply:
done